### ISSUE TITLE
[Proposal] Make Torch generic operators drop-in replacements for Tensor

### ISSUE THREAD

**Issue by Tenkei** [trained-by-humans/ml-pipes#70]
### Proposal type

Change

### Proposed owner

torch

### Problem and use case

I want to switch generic postprocess stages between NumPy and Torch by changing the imported module and the domain boundary, so I need `ml_pipes.torch` to expose the same generic operator names and configurations as `ml_pipes.tensor`.

Today, `Torch` mirrors `Tensor` operators, but their `Torch...` names require a separate pipeline spelling (`TorchArgMax`, `TorchSoftmax`, `TorchFilterTensors`, and so on). This makes backend-swappable pipelines harder to read, generate, and maintain despite the Torch package already documenting itself as a Tensor-postprocess mirror.

### Current workaround or gap in existing surface

I currently need to maintain backend-specific imports and operator names throughout otherwise equivalent pipelines:

```python
from ml_pipes.tensor import ArgMax, Softmax
# versus
from ml_pipes.torch import TorchArgMax, TorchSoftmax
```

The constructors of all the generic mirrors already have compatible parameter names and defaults, but their public names differ. Some output details also differ today, including `ArgMax` index dtype and tie ordering in selection/sorting operations.

### Proposed public surface

Expose the generic Tensor-mirror operators from `ml_pipes.torch` under their unprefixed canonical names:

```python
from ml_pipes.core import Pipeline
from ml_pipes.torch import ArgMax, GatherScores, Slice, Softmax, ToTorch

pipeline = Pipeline([
    ToTorch(device="cuda:0"),
    Softmax("class_logits", as_="class_probs"),
    Slice("class_probs", slice(None, -1)),
    ArgMax("class_probs", as_="classes"),
    GatherScores("class_probs", "classes", as_="scores"),
])
```

The Torch package should export unprefixed counterparts for all generic Tensor operators, including dtype/shape, selection/masking, mapping/math, sorting, and `Collate`.

Torch-specific values and boundary/device operators remain explicitly named, including `TorchTensorPayload`, `TorchTensorRegistry`, `ToTorch`, `ToNumpy`, `ToDevice`, and `TorchSynchronizeTensors`.

The generic operators should also share a documented conformance contract:

- identical constructor parameter names and defaults;
- `int64` index results across backends;
- deterministic first-input tie behavior for `TopK`, `TopKIndices2D`, and `SortTensorsBy`;
- equivalent outputs, shapes, dtypes, empty-input behavior, and validation errors, allowing normal floating-point tolerance where needed.

Out of scope: Torch runtime operators (`TorchInfer`, `TorchExtract`, `TorchDistribute`), Torch-native Vision postprocess, and NumPy/Torch conversion/value-type renames. Those can be proposed separately.

**Comment by Tenkei** [comment]
First step is done, next need to add cross-backend conformance tests and align index dtype, tie ordering, empty inputs, and validation errors where the implementations differ.
